---
int32 result
uint8 value_r
uint8 value_g
uint8 value_b
